File: Kotlin/DLNASam/app/src/main/java/com/example/dlnasam/BaseActivity.java
package com.example.dlnasam;

import android.app.Activity;
import android.os.Bundle;

import com.didi365.dlnasupport.RenderApplication;

public class BaseActivity extends Activity{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        RenderApplication.onCatchError(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    protected void onPause() {
        super.onPause();

        RenderApplication.onPause(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        RenderApplication.onResume(this);
    }
}

File: kotlinshop/BaseLibrary/src/main/java/com/example/chen/baselibrary/ui/activity/BaseActivity.kt
package com.example.chen.baselibrary.ui.activity

import android.os.Bundle
import android.view.View
import android.widget.FrameLayout
import com.example.chen.baselibrary.common.AppManager
import com.trello.rxlifecycle.components.support.RxAppCompatActivity
import org.jetbrains.anko.find


open class BaseActivity : RxAppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppManager.instarnce.addActivity(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        AppManager.instarnce.finishActivity(this)
    }

    //获取Window中视图content
    val contentView: View
        get() {
            val content = find<FrameLayout>(android.R.id.content)
            return content.getChildAt(0)
        }
}